### PR TITLE
fix(settings): Disallow project members to reset project API key

### PR DESCRIPTION
**Pull request by macobo**
After this change:
- /reset_token endpoint is only reachable by organization ADMINs or higher
- We hide the reset action for non-admins

## How did you test this code?

- See unit test
- Manual testing via shell_plus

### CODE REVIEW

File: posthog/api/team.py [hunk 1]
```diff
@@ -183,9 +183,10 @@ def get_permissions(self) -> List:
         Special permissions handling for create requests as the organization is inferred from the current user.
         """
         base_permissions = [permission() for permission in self.permission_classes]
+
+        # Return early for non-actions (e.g. OPTIONS)
         if self.action:
-            # Return early for non-actions (e.g. OPTIONS)
-            if self.action == "create":
+            if self.action in ("create", "reset_token"):
```
Comment (Twixes) [inline posthog/api/team.py:189]: `reset_token` should be `TeamMemberStrictManagementPermission` and not `OrganizationAdminWritePermissions`, as project-specific admins should have the same edit capabilities as org admins.

**Review by Twixes — CHANGES_REQUESTED** [review]

**Comment by Twixes** [comment]
This makes sense though I wonder what's the motivation for making the change now?

File: posthog/api/team.py (hunk 1 above)
Comment (macobo) [reply posthog/api/team.py:189]: Thanks for clarifying this!

**Comment by macobo** [comment]
> This makes sense though I wonder what's the motivation for making the change now?

Support hero this week, picking up odds and ends that users are pointing out. :)

**Review by Twixes — APPROVED** [review]
Looks good

**Pull request merged**